src/prometheus/alert.rs

```diff
@@ -0,0 +1,64 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct Alerts {
+    pub groups: Vec<AlertGroup>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AlertGroup {
+    pub name: String,
+    pub rules: Vec<AlertRules>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct AlertRules {
+    pub alert: String,
+    pub expr: String,
+    #[serde(rename = "for")]
+    pub for_: String,
+    pub labels: HashMap<String, String>,
+    pub annotations: HashMap<String, String>,
```

Comment:
Given that we're not exposing this publically, I wonder if it's worth making annotations a struct with summary and description fields?

Reply:
I implemented this module intended to be the final thing we produce and serialise for Prometheus (in our MVP, `rule-watcher` ConfigMaps) to ingest. So the whole structure put together maps exactly to a Prometheus alert rule: https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/